src/TNetwork.cpp

```diff
@@ -31,6 +31,63 @@
 
 typedef boost::asio::detail::socket_option::integer<SOL_SOCKET, SO_RCVTIMEO> rcv_timeout_option;
 
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <chrono>
+#include <mutex>
+
+std::unordered_map<std::string, std::vector<std::chrono::time_point<std::chrono::high_resolution_clock>>> connectionAttempts;
+std::mutex connectionAttemptsMutex;
+
+bool WatchingConnecting::IsConnectionAllowed(const std::string& clientAddress) {
+        // we check if there is an IP in the blocked list
```

Comment:
please run `clang-format` on all files you modified, for example `clang-format -i --style=file src/*.cpp include/*.h`. This will fix all the indentation and other tiny things :)